src/ImageSharp/Formats/Jpeg/Components/Block8x8F.Vector256.cs

```diff
@@ -0,0 +1,191 @@
+/// </content>
+internal partial struct Block8x8F
+{
+    /// <summary>
+    /// A number of rows of 8 scalar coefficients each in <see cref="Block8x8F"/>
+    /// </summary>
+    public const int RowCount = 8;
+
+#pragma warning disable SA1310 // Field names should not contain underscore
+    [FieldOffset(0)]
+    public Vector256<float> V256_0;
+    [FieldOffset(32)]
+    public Vector256<float> V256_1;
+    [FieldOffset(64)]
+    public Vector256<float> V256_2;
+    [FieldOffset(96)]
+    public Vector256<float> V256_3;
+    [FieldOffset(128)]
+    public Vector256<float> V256_4;
+    [FieldOffset(160)]
+    public Vector256<float> V256_5;
+    [FieldOffset(192)]
+    public Vector256<float> V256_6;
+    [FieldOffset(224)]
+    public Vector256<float> V256_7;
+#pragma warning restore SA1310 // Field names should not contain underscore
+
+    /// <summary>
+    /// <see cref="Vector256{Single}"/> version of <see cref="NormalizeColorsInPlace(float)"/> and <see cref="RoundInPlace()"/>.
+    /// </summary>
+    /// <param name="maximum">The maximum value to normalize to.</param>
+    [MethodImpl(InliningOptions.ShortMethod)]
+    public void NormalizeColorsAndRoundInPlaceVector256(float maximum)
+    {
+        Vector256<float> off = Vector256.Create(MathF.Ceiling(maximum * 0.5F));
+        Vector256<float> max = Vector256.Create(maximum);
+
+        this.V256_0 = NormalizeAndRoundVector256(this.V256_0, off, max);
+        this.V256_1 = NormalizeAndRoundVector256(this.V256_1, off, max);
+        this.V256_2 = NormalizeAndRoundVector256(this.V256_2, off, max);
+        this.V256_3 = NormalizeAndRoundVector256(this.V256_3, off, max);
+        this.V256_4 = NormalizeAndRoundVector256(this.V256_4, off, max);
+        this.V256_5 = NormalizeAndRoundVector256(this.V256_5, off, max);
+        this.V256_6 = NormalizeAndRoundVector256(this.V256_6, off, max);
+        this.V256_7 = NormalizeAndRoundVector256(this.V256_7, off, max);
+    }
+
+    /// <summary>
+    /// Loads values from <paramref name="source"/> using extended AVX2 intrinsics.
+    /// </summary>
+    /// <param name="source">The source <see cref="Block8x8"/></param>
+    public void LoadFromInt16ExtendedAvx2(ref Block8x8 source)
+    {
+        DebugGuard.IsTrue(
+            Avx2.IsSupported,
+            "LoadFromUInt16ExtendedAvx2 only works on AVX2 compatible architecture!");
+
+        ref short sRef = ref Unsafe.As<Block8x8, short>(ref source);
+        ref Vector256<float> dRef = ref Unsafe.As<Block8x8F, Vector256<float>>(ref this);
+
+        // Vector256<ushort>.Count == 16 on AVX2
+        // We can process 2 block rows in a single step
+        Vector256<int> top = Avx2.ConvertToVector256Int32(Vector128.LoadUnsafe(ref sRef));
+        Vector256<int> bottom = Avx2.ConvertToVector256Int32(Vector128.LoadUnsafe(ref sRef, (nuint)Vector256<int>.Count));
+        dRef = Avx.ConvertToVector256Single(top);
+        Unsafe.Add(ref dRef, 1) = Avx.ConvertToVector256Single(bottom);
```

Comment:
This one becomes stylistic preference, but you can freely mix the xplat APIs and the platform specific intrinsics.

That is, while you're wanting to use `V256 Avx2.ConvertToVector256Int32(V128)` instead of `V256.WidenLower/WidenUpper` for efficiency, you can just use `V256.ConvertToSingle()` still instead of `Avx.ConvertToVector256Single` since it is a 1-to-1 mapping.

-- As a note to myself, it would likely be beneficial to have `V256.Widen(V128)` APIs or similar; or to pattern match `V256.WidenLower(V256)` followed by `V256.WidenUpper(V256)`; so devs don't need to use platform specific intrinsics in such cases

Reply:
I didn't bother porting the existing Avx code as it worked but I might still do it.